tests/moduletool/test_module_tool_basics.py

```diff
@@ -332,3 +334,47 @@ async def test_module_requires_legacy_invalid(inmanta_module):
     )
     with pytest.raises(InvalidMetadata, match="Invalid legacy requires"):
         module.Module(None, inmanta_module.get_root_dir_of_module())
+
+
+def test_project_repo_type_module_v2(modules_dir, modules_repo, caplog):
+    """
+    Tests that repos that are strings and repos that are dict with
+    type 'git' are accepted and that repos with another type
+    will raise a warning. (issue #3565)
+    """
+    makeproject(modules_repo, "project", [], [])
+    projectdir = os.path.join(modules_repo, "project")
```

Comment:
I believe `makeproject` returns this path. You might be able to merge these two lines. No strict change request.